net/pfSense-pkg-pfBlockerNG-devel/files/usr/local/pkg/pfblockerng/pfblockerng.inc

```diff
@@ -565,7 +565,24 @@ function pfbng_text_area_decode($text, $mode=FALSE, $type=TRUE) {
 	if (!empty($customlist)) {
 		foreach ($customlist as $line) {
 			if (substr(trim($line), 0, 1) != '#' && !empty($line)) {
-
+				if ($idn) {
+					if (strpos($line, '#') !== FALSE) {
+						$tmpline = preg_split('/(?=#)/', $line);
+						if (substr($tmpline[0], 0, 1) == '.') {
+							// idn_to_ascii() returns empty string if it starts with '.' 
+							$tmpline[0] = '.' . idn_to_ascii(ltrim($tmpline[0], '.'));
```

Comment:
Need to validate that the idn_to_ascii() function doesn't return an empty line.

That condition would cause the whitelist to have a line with only the  ".", and that would probably whitelist all entries in DNSBL :)

Reply:
I see,
Fixed